File: in_array/in_array.py

# ATOMIC_FUNCTION: in_array(value,array)
# TAGS: list
# LANG:  python
# FUNCTION: in_array(value,array)
def in_array(value,array):
	return value in array

start_val_1 = ["hello","world"]
start_val_2 = "hello"
expected_end_val = True

end_val = in_array(start_val_2,start_val_1)

# TEST
if(end_val == expected_end_val):
    print("PASS")
else:
    print("FAIL")

File: in_array/in_array.js
// ATOMIC_FUNCTION: in_array(needle, haystack)
// TAGS: array
// LANG:  node.js
// FUNCTION: in_array(needle, haystack)
function in_array(needle, haystack) {
    var length = haystack.length;
    for(var i = 0; i < length; i++) {
        if(haystack[i] == needle) return true;
    }
    return false;
}
var start_val_1 = ["hello","world"];
var start_val_2 = "hello";
var expected_end_val = true;
var end_val = in_array(start_val_2,start_val_1);

// TEST
if(end_val==expected_end_val){
    console.log('PASS');
}else{
	
    console.log('FAIL');
}
